riscv-c-api.md

```diff
@@ -170,6 +170,90 @@ For example:
 * `__attribute__((interrupt("supervisor")))`
 * `__attribute__((interrupt("machine")))`
 
+### `__attribute__((target("<ATTR-STRING>")))`
+
+`target` attribute used for enable or disable a set of features or extensions
+for the fucntion.
+
+For instance, you can enable the `v` extension for a specific function even if
+`-march` or `-mcpu` does not include the `v` extension. Additionally,
+this won't alter the global setting.
+```c
+__attribute__((target("arch=+v")))
+int
+foo (int a)
+{
+  return a + 5;
+}
+```
+
+Function with target attribute should not affect the file scope build
+attributes, e.g. File compiled with `-march=rv64ima` and a function declares with
+`__attribute__((target("arch=+zbb")))`, the `Tag_RISCV_arch` build attribute
+should be `rv64ima` rather than `rv64ima_zbb`.
+
+The compiler may emit mapping symbol at the beginning of the function with
+target attribute if the function use different set of ISA extension.
+
+`<ATTR-STRING>` can be specify following target attributes:
+
+- `arch=`: Adding extra extensions, omit certain extension, or specific
+           extension setting for the function.
+- `tune=`: Specify the pipeline mode and cost model with specific
+           microarchitecture or core for the function.
+- `cpu=`: Specify the pipeline mode, cost model and extension setting for the
+          function.
+
+The interactions among `arch`, `tune`, and `cpu` are the same as the `-march`,
+`-mtune`, and `-mcpu` options. The `cpu` can be treated as a combination of
+`arch` + `tune` but has a lower priority than the other two. For instance,
+`cpu=sifive-u74` is equivalent to `arch=rv64gc` and `tune=sifive-7-series`.
+However, if `arch=` or `tune=` values are provided, they will override the `cpu`
+value. Therefore, `cpu=sifive-u74;arch=rv64g` will be equivalent to
+`arch=rv64g;tune=sifive-7-series`, and `cpu=sifive-u74;tune=sifive-5-series`
+will be equivalent to `arch=rv64gc;tune=sifive-5-series`.
+
+If the same type of attribute is given more than once, only the latest one
+takes effect, e.g. `arch=+zbb;arch=+zba` will be equivalent to `arch=+zba`.
```

Comment:
What is the effect of having -march at the command line and a march on the function which is a subset of the command line?  I think either disallowing or full replacement is fine, we just don't want to try merging.

If I specific march at the command line, and mcpu via the attribute, what happens?  mcpu would normally be lower priority, but is this still true comparing command line and attribute?

Reply:
Add one more paragraph to describe the interaction between attribute and option.

`arch=` has two different behavior here: full-arch-str will override `-march`, but `+ext` will combine with `-march`.
`cpu=` will override `-march` if `arch=` not given.

Reply:
Drop for now